src/Entities/Minecart.cpp

```diff
@@ -1206,10 +1206,19 @@ void cMinecartWithChest::OpenNewWindow()
 
 void cMinecartWithChest::Destroyed()
 {
-	GetWindow()->OwnerDestroyed();
+	if (GetWindow() != nullptr)
+	{
+		GetWindow()->OwnerDestroyed();
+	}
 	cItems Pickups;
 	m_Contents.CopyToItems(Pickups);
-	GetWorld()->SpawnItemPickups(Pickups, GetPosX(), GetPosY() + 1, GetPosZ(), 4);
+
+
+	// This makes the command not execute if the world is in the midst of destruction :)
+	GetWorld()->ScheduleTask(1, [this, &Pickups](cWorld & World)
```

Comment:
You are right. Reliability is number 1 priority. But I just can't help but feel that the logic around how this works (esp. cases like shutting down - who accesses what in which destructor for example) is very fuzzy, and that's causing the bugs/crashes.